### PR TITLE
ddl/callback_test.go: migrate test-infra to testify

### LINKED ISSUE #28629: migrate test-infra to testify for `ddl/callback_test.go`

### PR DESCRIPTION
**Pull request by zach030**
### What problem does this PR solve?

Issue Number: close #28629 

Problem Summary:

### What is changed and how it works?
Uses testify for ddl/callback_test.go
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/callback_test.go [hunk 1]
```diff
@@ -117,9 +118,9 @@ func (tc *TestDDLCallback) OnWatched(ctx context.Context) {
 	tc.BaseCallback.OnWatched(ctx)
 }
 
-func (s *testDDLSuite) TestCallback(c *C) {
+func (s *testDDLSuite) TestCallback(t *testing.T) {
```
Comment (hawkingrei) [inline ddl/callback_test.go:121]: It should look like this.

```
func TestCallback(t *testing.T)
```

**Comment by hawkingrei** [comment]
Thanks for your PR!

File: ddl/callback_test.go (hunk 1 above)
Comment (zach030) [reply ddl/callback_test.go]: fixed

**Comment by hawkingrei** [comment]
@tisonkun @djshow832 @tangenta PTAL

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by zimulala** [comment]
PTAL @hawkingrei

**Review by hawkingrei — APPROVED** [review]
LGTM

**Pull request merged**